libraries/AP_HAL/hwdef/scripts/hwdef.py

```diff
@@ -301,16 +305,21 @@ def assert_good_define(self, name):
     def process_line_define(self, line, depth, a):
         # defines are currently written out a little haphazardly, but
         # we do the sanity checks here:
-        result = re.match(r'define\s*([A-Z_0-9]+)\s*', line)
-        if result is not None:
-            # ensure that stale defines aren't introduced into hwdef files:
-            name = result.group(1)
-            self.assert_good_define(name)
+        result = re.match(r'define\s*([A-Z_0-9]+)\s*(?:\s*(.*))?', line)
```

Comment:
```suggestion
        result = re.match(r'define\s*([A-Z_0-9]+)\s*(.*)', line)
```

Duplicating the `\s*` then making sure we don't capture it is unnecessary. `.*` can never fail to produce a match anyway